router/router.go

```diff
@@ -1346,17 +1282,28 @@ func (rt *HandleT) getWorkerPartition(userID string) int {
 	return misc.GetHash(userID) % rt.noOfWorkers
 }
 
-func (rt *HandleT) shouldThrottle(destID, userID string, throttledAtTime time.Time) (canBeThrottled bool) {
-	if !rt.throttler.IsEnabled() {
-		return false
+func (rt *HandleT) shouldThrottle(job *jobsdb.JobT, parameters JobParametersT) (
+	throttling.TokenReturner,
+	bool,
+) {
+	throttler := rt.getThrottler(parameters.DestinationID)
+	if throttler == nil {
+		return nil, false
+	}
+
+	throttlingCost := rt.getThrottlingCost(parameters.DestinationID, job)
+
+	limited, tokensReturner, err := throttler.CheckLimitReached(throttlingCost)
+	if err != nil {
+		// we can't throttle, let's hit the destination, worst case we get a 429
+		return nil, false
 	}
 
-	// No need of locks here, because this is used only by a single goroutine (generatorLoop)
-	limitReached := rt.throttler.CheckLimitReached(destID, userID, throttledAtTime)
-	if !limitReached {
-		rt.throttler.Inc(destID, userID, throttledAtTime)
+	if !limited {
+		return tokensReturner, false
 	}
-	return limitReached
+
+	return nil, true
```

Comment:
can't we simplify this, e.g.

```
return tokensReturner, !limited
```
Will `throttler.CheckLimitReached` return a non-nil tokensReturner if limit is reached? It shouldn't right?